src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,175 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exceptions.RegistrationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final String VALID_LOGIN = "validLogin";
+    private static final String VALID_PASSWORD = "validPassword";
+    private static final String EXISTING_USER_LOGIN = "existingUserLogin";
+    private static final String SHORT_LOGIN = "log";
+    private static final String SHORT_PASSWORD = "pass";
+    private static final String EMPTY_STRING = "";
+    private static final int VALID_AGE = 24;
+    private static final int MAX_VALID_AGE = 100;
+    private static final int INVALID_AGE = 15;
+    private static final int MAX_LOGIN_LENGTH = 25;
+    private static final int MAX_PASSWORD_LENGTH = 25;
+    private static final String TOO_LONG_LOGIN = "a".repeat(MAX_LOGIN_LENGTH + 1);
+    private static final String TOO_LONG_PASSWORD = "a".repeat(MAX_PASSWORD_LENGTH + 1);
+
+    private RegistrationService registrationValidator;
+    private StorageDao storageDao;
+
+    @BeforeEach
+    void setUp() {
+        Storage.people.clear();
+        registrationValidator = new RegistrationServiceImpl();
```

Comment:
it's better to initialise RegistrationServiceImpl and StorageDaoImpl in BeforeAll method

and Storage.people.clear(); in AfterEach